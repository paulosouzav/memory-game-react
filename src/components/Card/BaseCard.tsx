import styled, { css } from 'styled-components';
import { transitions } from 'polished';
import { SHADOWS } from '../../constants/shadows';
import { ROTATION } from '../../constants/rotation';
import { mediaMax } from '../../utils/mediaQuery';
import Face from './Face';

export interface IBaseCard {
  isFlipped?: boolean;
  isMatched?: boolean;
}

const BaseCard: any = styled.div`
  width: 155px;
  height: 210px;
  margin: 10px;
  position: relative;
  border-radius: 8px;

  ${transitions(['transform', 'opacity'], '.4s')}
  transform: scale(1);
  transform-style: preserve-3d;

  ${SHADOWS.normal}
  
  ${mediaMax.sm`
    width: 90px;
    height: 130px;
  `}

  ${({ isFlipped }: IBaseCard) => !isFlipped && ACTIVE}
  ${({ isMatched }: IBaseCard) => isMatched && MATCHED}
  ${({ isFlipped }: IBaseCard) => isFlipped && ROTATION.half}
`;

const ACTIVE = css`
  &:active {
    transform: scale(0.97);
  }
`;

const MATCHED = css`
  cursor: not-allowed;
`;

BaseCard.Face = Face;
BaseCard.displayName = 'BaseCard';

export default BaseCard;
