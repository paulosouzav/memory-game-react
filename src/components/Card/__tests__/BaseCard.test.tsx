import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import BaseCard from '../BaseCard';

describe('BaseCard', () => {
  it('should render without failing', () => {
    const wrapper = mount(<BaseCard />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have element Face', () => {
    expect(BaseCard.Face).toBeDefined();
  });
});
