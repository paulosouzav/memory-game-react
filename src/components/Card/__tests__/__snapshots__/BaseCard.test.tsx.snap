// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCard should render without failing 1`] = `
.c0 {
  width: 155px;
  height: 210px;
  margin: 10px;
  position: relative;
  border-radius: 8px;
  -webkit-transition: -webkit-transform .4s,opacity .4s;
  -webkit-transition: transform .4s,opacity .4s;
  transition: transform .4s,opacity .4s;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  box-shadow: 0 0 8px rgba(0,0,0,0.5);
}

.c0:active {
  -webkit-transform: scale(0.97);
  -ms-transform: scale(0.97);
  transform: scale(0.97);
}

@media (max-width:576px) {
  .c0 {
    width: 90px;
    height: 130px;
  }
}

<BaseCard>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "Face": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "rules": Array [
              "
  width: 100%;
  height: 100%;
  object-fit: contain;
  padding: 20px;
  border-radius: 8px;
  position: absolute;
  background: #ecf0f1;

  transition: opacity 0.4s;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;

  ",
              [Function],
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "Face",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  width: 155px;
  height: 210px;
  margin: 10px;
  position: relative;
  border-radius: 8px;

  ",
            "transition: transform .4s, opacity .4s;",
            "
  transform: scale(1);
  transform-style: preserve-3d;

  ",
            "
    box-shadow: 0 0 8px ",
            "rgba(0,0,0,0.5)",
            ";
  ",
            "
  
  ",
            "
      @media (",
            "max",
            "-width: ",
            "576",
            "px) {
        ",
            "
    width: 90px;
    height: 130px;
  ",
            ";
      }
    ",
            "

  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "BaseCard",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</BaseCard>
`;
